cluster-autoscaler/core/scale_down_node_processor.go

```diff
@@ -0,0 +1,83 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package core
+
+import (
+	"reflect"
+
+	apiv1 "k8s.io/api/core/v1"
+	"k8s.io/klog"
+
+	"k8s.io/autoscaler/cluster-autoscaler/context"
+	"k8s.io/autoscaler/cluster-autoscaler/utils/errors"
+)
+
+// DefaultScaleDownNodeProcessor filters out scale down candidates from nodegroup with
+// size <= minimum number of nodes for that nodegroup
+type DefaultScaleDownNodeProcessor struct {
```

Comment:
Can we use a more descriptive name? Maybe something along the lines of PreFilteringScaleDownNodeProcessor?

Other default processors have such names, example: https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/processors/status/scale_up_status_processor.go#L79

Reply:
Done